tests/multithreading/thread_local_storage_08.output

```diff
@@ -0,0 +1,8 @@
+
+DEAL::Creating
+DEAL::Copying
+DEAL::Copying
+DEAL::Copying
+DEAL::Copying
+DEAL::Copying
+DEAL::Copying
```

Comment:
```suggestion
DEAL::Moving
DEAL::Moving
DEAL::Moving
DEAL::Moving
DEAL::Moving
DEAL::Moving
```

Reply:
The `exemplar` is used to copy its value into the thread-local `data` if `data` is still empty on a thread. If the `exemplar` is empty, new instances of `X` are constructed instead, using the default ctor.

Without the bugfix in this PR the `exemplar` is not copied/moved and the threads in this test would "create" new instances of `X` instead of copy-constructing them.